src/game/client.cpp

```diff
@@ -477,11 +477,15 @@ namespace client
                 items.emplace_back(v >= W_OFFSET && v < W_ITEM ? v : 0);
             }
         }
-        game::player1.loadweap.shrink(0);
-        loopv(items) if(game::player1.loadweap.find(items[i]) < 0)
-        {
-            game::player1.loadweap.add(items[i]);
-            if(game::player1.loadweap.length() >= W_LOADOUT) break;
+        game::player1.loadweap.clear();
+        for (const auto i : items) {
+            if (std::find(game::player1.loadweap.begin(),
+                game::player1.loadweap.end(), i) == game::player1.loadweap.end()) // NOT found
+            {
+                game::player1.loadweap.emplace_back(i);
+                if (game::player1.loadweap.size() >= W_LOADOUT)
```

Comment:
Please wrap all conditional code (e.g., branches of an `if`) in braces to create a proper scope. It is a common issue in refactorings to add lines to existing code without realizing that the scope had ended at the semicolon. This can be avoided by always using a scope.

Reply:
Would this work?:
```c++
if (std::none_of(game::player1.loadweap.begin(),
                game::player1.loadweap.end(), [=](auto d) { return d == i; }))
```
I hope there isn't too much overhead if done like this.

Reply:
> This can be avoided by always using a scope.

That is silly and adds unnecessary LoC; even the Kernel's coding style agrees:
> Do not unnecessarily use braces where a single statement will do.

Reply:
Just because the Linux kernel uses it doesn't mean it's beginner friendly. Quite the opposite, really. I've seen this kind of problem too often, so I always use a scope now. It's a pre-emptive measure used to prevent branching problems.

"Saving lines of code" is an argument which I often heard before, but it is not a good one. It might have been relevant, back in the days when there were just 12 lines on one screen. Compacting the code too much like this does not aid readability, quite the opposite, and is often used to compact and therefore kind of "hide" complexity. Look at this line, it's probably too long already.

Python for instance, while not everyone likes the lack of braces, does this very well in general. Whenever the code branches, moving the conditional code on a new line (and therefore adding some indentation as well) is very strongly recommended. You _could_ use the same line, but this is ugly and cumbersome. See https://docs.python-guide.org/writing/style/#one-statement-per-line.

Regarding braces, many popular code styles demand them, there are just few exceptions, mostly for old projects who had established such a style long ago. Not using braces "saves" 1-2 lines depending on the code style. I personally prefer to move the opening one on the same line as the conditional/loop statement. As stated in https://github.com/blue-nebula/base/pull/248#discussion_r917471888, this ensures that on future refactorings, new code won't accidentally be added in the hope that it will be conditional as well. Furthermore, this also aids the readability of the code. You can quickly see where branching happens. New variables have a clear scope. You can't be surprised when commenting out lines. You do not have to be careful about the semicolon being the end of the conditional code block.

This is not an obfuscated code competition. The code should be easy to read. Branches should be easily visible. Code should be written with the need for future changes in mind.